install_config/persistent_storage/persistent_storage_csi.adoc

```diff
@@ -0,0 +1,358 @@
+storage].
+
+
+[NOTE]
+====
+CSI volumes is a beta feature and may change in a future release of {product-title}. Due to its beta status, it does not support SELinux yet.
+====
+
+[NOTE]
+====
+{product-title} does not ship with any CSI drivers. It is recommended to use the CSI drivers provided by link:https://kubernetes-csi.github.io/docs/Drivers.html[community or storage vendors].
+====
+
+[NOTE]
+====
+{product-title} {product-version} supports version 0.2.0 of link:https://github.com/container-storage-interface/spec[CSI specification].
+====
+
+
+[[install-config-persistent-storage-csi-architecture]]
+== Architecture
+
+CSI drivers are typically shipped as container images. These containers are not aware of {product-title} where they run. To use CSI-compatible storage backed in {product-title}, the cluster administrator must deploy several components that serve as a bridge between {product-title} and the storage driver.
+
+The following diagram provides high-level overview about these components running in pods in {product-title} cluster.
+
+image::OpenShift_CSI_Arch_470752_0518.png["Architecture of CSI components"]
+
+It is possible to run multiple CSI drivers for different storage backends. Each driver needs its own external controllers deployment, and DaemonSet with the driver and CSI registrar.
+
+=== External CSI Controllers
+
+External CSI Controllers is a deployment that deploys one or several pods with two containers:
+
+* External CSI attacher container that translates `attach` and `detach` calls from {product-title} to respective `ControllerPublish` and `ControllerUnpublish` calls to CSI driver.
+* External CSI provisioner container that translates `provision` and `delete` calls from {product-title} to respective `CreateVolume` and `DeleteVolume` calls to CSI driver.
+* CSI driver container.
+
+The CSI attacher and CSI provisioner containers talk to the CSI driver container using UNIX Domain Sockets, ensuring that no CSI communication leaves the pod. The CSI driver is not accessible from outside of the pod.
+
+[NOTE]
+====
+`attach`, `detach`, `provision` and `delete` operations typically require the CSI driver to use credentials to the storage backend. Run the CSI controller pods on xref:../../install_config/install/advanced_install.adoc#configuring-dedicated-infrastructure-nodes[infrastructure nodes] so the credentials never leak to user processes, even in the event of a catastrophic security breach on a compute node.
+====
+
+[NOTE]
+====
+The external attacher must also run for CSI drivers that do not support third-party attach/detach operations. The external attacher will not issue any `ControllerPublish` or `ControllerUnpublish` operations to CSI driver. However, it still must run to implement the necessary {product-title} attachment API.
+====
```

Comment:
s/operations to CSI driver/operations to the CSI driver

Reply:
fixed